static_compress/mixin.py

```diff
@@ -71,47 +72,64 @@ def get_modified_time(self, name):
         return self._datetime_from_timestamp(getmtime(alt))
 
     def post_process(self, paths, dry_run=False, **options):
-        if hasattr(super(), "post_process"):
-            yield from super().post_process(paths, dry_run, **options)
-
         if dry_run:
             return
 
+        files = OrderedDict()
+
+        super_class = super()
+        if hasattr(super_class, "post_process"):
+            generator = super_class.post_process(paths, dry_run, **options)
+        else:
+            generator = iter((f, f, False) for f in paths)
+
+        for file, hashed_file, processed in generator:
+            files[file] = hashed_file
+            yield file, hashed_file, processed
+
         for name in paths.keys():
-            if not self._is_file_allowed(name):
+            if not (name in files and self._is_file_allowed(name)):
```

Comment:
`if name not in files or not self._is_file_allowed(name)`

Reply:
that's a matter of taste :)